From: Kai Liu <kai.liu@suse.com>
Date: Sun, 1 Jan 2022 18:32:26 +0800
Subject: mm: fix compile issue of mm/mempolicy.c
Patch-mainline: Never, fixes backported stable patch
References: bsn#19

The stable commit ee6f34215c5d in v5.10.89 backported mainline commit
338635340669, but replaced __alloc_pages() with __alloc_pages_nodemask()
as the former is not available in v5.10. However the situation is quite
the opposite in SEL-2.0 as we have backprted related mainline code in
the below patch:

patches.mainline/v5.13-rc1-mm-page_alloc-combine-__alloc_pages-and-__alloc_page.patch

Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 mm/mempolicy.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mm/mempolicy.c
+++ b/mm/mempolicy.c
@@ -2265,8 +2265,7 @@ struct page *alloc_pages_vma(gfp_t gfp,
 			 * memory with both reclaim and compact as well.
 			 */
 			if (!page && (gfp & __GFP_DIRECT_RECLAIM))
-				page = __alloc_pages_nodemask(gfp, order,
-							hpage_node, nmask);
+				page = __alloc_pages(gfp, order, hpage_node, nmask);
 
 			goto out;
 		}
